### DISCUSSION TITLE
How to Download the VHR Dataset

### DISCUSSION THREAD

**Discussion by osgeokr** [torchgeo/torchgeo#1944] [Q&A] [answered]
I'm attempting to practice Object Detection using only the Positive image set from the VHR-10 dataset.
Following the guidelines on [torchgeo.datasets](https://torchgeo.readthedocs.io/en/stable/api/datasets.html#vhr-10), I've installed additional libraries pycocotools and rarfile. My torchgeo version is 0.5.2.
```
%pip install -q -U torchgeo
%pip install -q -U pycocotools
%pip install -q -U rarfile
```
However, when I try to download the VHR10 dataset with the code below, I encounter an error message stating `NotRarFile: Not a RAR file`. Could you let me know what I might be missing? Thank you in advance for your response.
```
from torchgeo.datasets import VHR10
ds = VHR10(root='data/VHR10/', split='positive', download=True)
```

**Comment by adamjstewart** [comment]
@ashnair1 can you confirm that the download still works properly? @osgeokr can you add `checksum=True` to verify if you're getting the expected checksum?

P.S. You can also install all optional dataset dependencies using `pip install torchgeo[datasets]`

**Comment by ashnair1** [comment]
It's working on my end. 

@osgeokr Can you post the full stack trace?

**Comment by osgeokr** [comment]
@adamjstewart @ashnair1 During the attempt of [`pip install torchgeo[datasets]`](https://github.com/microsoft/torchgeo/discussions/662) as you suggested, I encountered an error due to the [zipfile-deflate64](https://pypi.org/project/zipfile-deflate64/) package not supporting Python 3.11. Consequently, I downgraded to Python 3.10 and proceeded with the reinstallation (`pip install torchgeo[datasets]`).

Afterwards, when I executed the following code, I encountered a ModuleNotFoundError: `No module named 'gdown' error`:
```
from torchgeo.datasets import VHR10
ds = VHR10(root='data/VHR10/', download=True, checksum=True)
```
I successfully downloaded the NWPU VHR-10 dataset.rar after installing the gdown package. However, during the decompression process, I encountered the `RarCannotExec: Cannot find working tool error`, and the compression did not unzip properly. For your information, I am using Windows 10 operating system. This is suspected to be an error with the rarfile package.

After installing the [aspose-zip](https://pypi.org/project/aspose-zip/) package additionally, I successfully decompressed the file using the following code:
```
import aspose.zip as az

with az.rar.RarArchive("data/VHR10/NWPU VHR-10 dataset.rar") as archive:
    archive.extract_to_directory("data/VHR10/")
```
Up to this point, these are the observations made in the local environment. I'm curious if there's a learning example of downloading the VHR10 dataset in a Colab environment. Thank you. :)

**Reply by adamjstewart** [reply, marked as answer]
Surprised zipfile-deflate64 doesn't have newer wheels yet, I commented on https://github.com/brianhelba/zipfile-deflate64/pull/41

The missing gdown is the result of https://github.com/pytorch/vision/pull/8237, which replaced requests with gdown but didn't add an explicit dependency on gdown. I wonder if we can convince them to add this as a required dependency...

It sounds like the root of the issue is that you don't have unrar installed. This is related to https://github.com/markokr/rarfile/issues/72. Basically, rarfile requires unrar, but doesn't install it automatically because it isn't a Python library and the license prevents them from bundling it. So you'll have to install unrar yourself using a package manager like choco or pacman or a binary download like https://gnuwin32.sourceforge.net/packages/unrar.htm.

**Reply by osgeokr** [reply]
I downloaded the [UnRAR for Windows](https://www.rarlab.com/rar_add.htm) and extracted the unrarw64.exe file.
After that, the following code worked successfully!
```
from torchgeo.datasets import VHR10
ds = VHR10(root='data/VHR10/', split='positive', download=True)
```
Here are the extra steps I confirmed in the Windows environment:
- Installing torchgeo via `pip install torchgeo[datasets]`
- Installing the `gdown` package
- Downloading and extracting the `unrarw64.exe` file

I think it would be helpful to add related information to the [VHR-10 Note](https://torchgeo.readthedocs.io/en/stable/api/datasets.html#vhr-10). 

Thank you for your advice! :)
